spec/configuration_spec.rb

```diff
@@ -1,44 +1,106 @@
 require 'spec_helper'
 
-RSpec.describe "Configuration" do
-  around do |spec|
-    Inky.configure do |config|
-      old = config.template_engine
-      spec.run
-      config.template_engine = old
+RSpec.describe "Inky.configuration" do
+  it "returns an Inky::Configuration object" do
+    expect(Inky.configuration).to be_an_instance_of Inky::Configuration
+  end
+
+  describe "=" do
+    it "accepts an Inky::Configuration" do
+      current_config = Inky.configuration
+      config = Inky::Configuration.new
+
+      expect { Inky.configuration = config }.to_not raise_error
+      expect(Inky.configuration).to_not eq(current_config)
+      expect(Inky.configuration).to eq(config)
+
+      expect { Inky.configuration = {} }.to raise_error(TypeError, "Not an Inky::Configuration")
+      expect(Inky.configuration).to eq(config)
+
+      expect { Inky.configuration = current_config }.to_not raise_error
```

Comment:
line 17 is to check that an error is raised with invalid input, 18 is to check taht the config was not modified when an error occured, and 20 is to verify that the error is indeed because the value was invalid

Reply:
By not useful I meant that it is not possible to succeed the last test and fail either of the first two. The last test encompasses the previous ones. But they don't hurt and serve as sanity checking.